lighthouse-core/report/html/renderer/report-renderer.js

```diff
@@ -193,8 +193,8 @@ class ReportRenderer {
     Util.i18n = i18n;
 
     const detailsRenderer = new DetailsRenderer(this._dom);
+    detailsRenderer.setLighthouseReport(report);
```

Comment:
the perf category renderer and a separate pr #10716 both need some data from the LHR to render, so let's put the report on the details renderer

done via setLighthouseReport so that the property will be optional and existing usages won't break (i.e. progressive enhancement)

Reply:
add a comment explaining why we're (somewhat) violating our layering.

Reply:
tbh I don't think that's a useful comment.

Reply:
it's useful for the same reason that led you to write this review comment. :p

here's what i was thinking

```
// The scorecalc link (_getScoringCalculatorHref()) and the elementScreenshotRenderer both need data in the LHR outside of what's provided
// With the report stored like this, the property is optional and existing usages won't break.
// TODO: Fix layering in the grand refactor of the ReportRenderer
```

Reply:
> // TODO: Fix layering in the grand refactor of the ReportRenderer

@paulirish Please
1. Create an issue
2. Assign someone
3. Assign it a priority

Grand refactors are an engineering smell, but incremental refactors tend to only happen if you do them when you actually need them. Otherwise they need to actually be scheduled.